### DISCUSSION TITLE
GitHub Access Token 저장 위치

### DISCUSSION THREAD

**Discussion by openmpy** [two-clock/backend#41] [General]
@jjangsky 님, GitHub Access Token 저장 위치에 대해서 생각을 다시 해봤습니다.
슬랙에서 말씀 해주셨던 방식은 ``JWT 토큰의 Claim 에 단방향 암호화를 진행해서 넣자`` 인데 이렇게 하면 문제점이 있습니다.
단방향 암호화를 진행하면 복호화를 진행할 수가 없어서 GitHub Access Token 에서 정보를 가져올 수가 없게 됩니다.
또 GitHub Access Token 은 민감한 정보라고 생각 됩니다. JWT 토큰의 Claim 에 저장하는 방식은 너무 위험한거 같습니다.

그래서 서버에 저장하는게 맞다고 생각합니다.
MySQL 보다는 빠른 접근이 가능한 Redis 에 양방향 암호화를 진행해서 저장하는게 맞다고 생각합니다.
@jjangsky 님은 어떻게 생각하시는지 답변 부탁드리겠습니다.
감사합니다.

**Comment by jjangsky** [comment]
@openmpy 맞습니다~
Access Token같은 귀중한 정보를 단방향으로 저장해서 언제든지 쉽게 사용할 수 있지만 사용자는 알 수 없게 설정하기 위한 생각이었는데
다시 생각해보니 단방향 암호화는 일치/불일치 에 대한 결과만 알 수 있다는 것을 망각하고 있었네요😢
(패스워드 검증 생각하고 이렇게 하면 안전하겠구나 ! 라는 착각...)

말씀하신대로 완벽한 암호화가 불가능 한 상황이면 인메모리 DB에 저장하는 것이 가장 효율적일것 같습니다 !

어제 확인해보니 Code로 발급되는 Access Token 자체가 만료기간이 따로 없는줄 알았는데
테스트 후, 다시 밥먹고 테스트 해보니까 특정 시간이 지나면 만료가 되는 것 같더라구여...
-> RestClient 요청 시점에 Exception에 걸리는것 확인하였습니다 !
이것 또한 추후 추가 로직이 필요할것 같습니다 !

**Reply by openmpy** [reply]
8시간으로 알고 있는데 한번 찾아봐야겠습니다.
답변 감사합니다 👍

**Reply by openmpy** [reply]
찾아본 결과 1시간이네요.. /refresh 엔드포인트로 요청 시 GitHub Access Token 까지 재발급 받는 로직을 추가 해두겠습니다.
